### PR TITLE
new: functions to tweak GregTech -> new version 1.6

### PR DESCRIPTION
**Pull request by zetti68**
Hello LionZXY,
I added some functions and changed a little bit to support the current version of GregTech 6.
Please see my changes below and in source code.

Details:
- use RecipesMap with keys to get recipes for machine
- add ZenMethods to add custom recipes also with fluid stacks
- save added recipes and remove it before /mt reload
- new method to remove recipes with equal inputs
- new command: /MTUtils GTkeys [1...n] to list all recipe keys in game [1...n -> first ten,...]
- new command: /MTUtils GTkey "machine name" to get recipe key

Best regards
Zetti68

### CODE REVIEW

**Comment by LionZXY** [comment]
Wow! Ok, I check PR today and apply or deny.

**Comment by zetti68** [comment]
Hello Lion, I found and corrected one big bug. My wish was to delete all added recipes before /mt reload. But I found only the scripting way to do this (MTUtilsGT.removeAddedRecipes();)

An event from minetweaker would be good to remove recipes automatically after start of /mt reload.

And I added the function with liquid arrays to iterate only one time over all recipes.
It works with the newest GT version (6.05.41) but not with older one. Gregorius changed a lot...

**Comment by LionZXY** [comment]
Ok. I update it in curse. It's ok?

**Comment by zetti68** [comment]
Thanks, that's ok.

I changed the fluid mapping function to a better scriptable form:
the old one was:
```
MTUtilsGT.addFluidInput(
  [<oldFluid1>,<oldFluid2><oldFluid3>], 
  [<newFluid1>,<newFluid2>,<newFluid3>]
);
```

new form:
```
MTUtilsGT.addFluidInput([
  [<oldFluid1>,<newFluid1>],
  [<oldFluid2>,<newFluid2>],
  [<oldFluid3>,<newFluid3>]
]);

```
I think, the new form is better to use in script, if you have a lot to map.

Additionally I added an example script in Wiki directory. May be it is helpful.

**Comment by LionZXY** [comment]
> Additionally I added an example script in Wiki directory. May be it is helpful.

That would be very cool.

**Comment by LionZXY** [comment]
Also I get some money from curse on Steam. It would be fair to buy a game for you. Which one would you like?

**Comment by zetti68** [comment]
Thanks, but not necessary. I'm in a full time job and program in Java. Look at my nickname and you will know my age (my birthday is in July). It is only my hobby. And I only play minecraft ;-).
I wanted to use minechem fluids in gregtech recipes and found your mod. The changes solve my problem and so I'm happy :-). That's enough for me. It would be nice if you will add me to the credits in mcmod.info.

**Comment by LionZXY** [comment]
Ok. Can you added example in Wiki?

File: src/main/java/ru/lionzxy/damagetweaker/mods/GregTechHandler.java [hunk 1]
```diff
@@ -1,50 +1,219 @@
...
+					fluidOutputs, aDuration, aEUt, 0L);
+
+			sRecipesCache.addRecipe(fieldName, recipe);
+
+			recipeMap.addRecipe(recipe, false, false, false);
+			System.out.println("[MTUtilsGT] Recipe for variable " + fieldName + " add!");
+			MineTweakerAPI.logInfo("[MTUtilsGT] Recipe for variable " + fieldName + " add!");
+		} catch (Exception e) {
+			MineTweakerAPI.logError(
+					"[MTUtilsGT] Not found variable " + fieldName + " in gregapi.recipes.Recipe.RecipeMap\n", e);
```
Comment (CanisArtorus) [inline src/main/java/ru/lionzxy/damagetweaker/mods/GregTechHandler.java:128]: Unhelpful message, could have been any error.

**Pull request merged**